rslaser/rspulse/pulse.py

```diff
@@ -44,59 +44,40 @@
 )
 
 
-def check_fields(input_obj, req_fields, exception, class_name):
-    for p in input_obj:
-        if p not in req_fields:
-            raise exception(f'invalid inputs: {p} is not a parameter to {class_name}')
-
-
-def _validate_params(input_params):
-    check_fields(input_params, _LASER_PULSE_DEFAULTS.keys(), InvalidLaserPulseInputError, 'LaserPulse')
-    check_fields(input_params.slice_params, _LASER_PULSE_SLICE_DEFAULTS.keys(), InvalidLaserPulseInputError, 'LaserPulseSlice')
+def check_fields(input_params, obj):
```

Comment:
It looks like this method is only called in `_validate_params`. I think the body of it can just be moved in there in the base class